### PR TITLE
refactor: Remove internal/utils/utils.go

### PR DESCRIPTION
**Pull request by michi-covalent**
- ParseCiliumVersion() doesn't have any logic in it. Directly call semver.ParseTolerant() instead.
- MustParseBool() is only used by the encryption package. Move it there and unexport it.

### CODE REVIEW

**Review by jschwinger233 — APPROVED** [review]

File: internal/helm/helm.go [hunk 1]
```diff
@@ -239,7 +238,7 @@ func ResolveHelmChartVersion(versionFlag, chartDirectoryFlag, repository string)
 }
 
 func resolveChartVersion(versionFlag string, repository string) (semver2.Version, *chart.Chart, error) {
-	version, err := utils.ParseCiliumVersion(versionFlag)
+	version, err := semver2.ParseTolerant(versionFlag)
```
Comment (derailed) [inline internal/helm/helm.go:241]: nit: import alias to semver (to be consistent) or semver4

**Review by derailed — CHANGES_REQUESTED** [review]
@michi-covalent Nice!

File: internal/helm/helm.go (hunk 1 above)
Comment (michi-covalent) [reply internal/helm/helm.go:241]: oh yeah i took care of it in another pull request here. let me rebase 🚀🙏

**Review by derailed — APPROVED** [review]
@michi-covalent Tx for the updates! Loving your early spring cleaning efforts!

**Review by derailed — APPROVED** [review]
@michi-covalent LGTM

**Pull request merged**